### PR TITLE
✨Add form submit service

### PR DESCRIPTION
**Pull request by calebcordry**
This was introduced so that extra `<input hidden>` elements could be added to a form before it will be submitted. Specifically `Linker` would like the ability to append any resolved linkers that may exist.

### CODE REVIEW

File: extensions/amp-form/0.1/form-submit-service.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+
+export class FormSubmitService {
+  /**
+   * Global service used to register callbacks we wish to execute when an
+   * amp-form is submitted.
+   */
+  constructor() {
+    /** @private {!Array} */
+    this.callbacks_ = [];
```
Comment (lannka) [inline extensions/amp-form/0.1/form-submit-service.js]: we can use observable.js

File: extensions/amp-form/0.1/form-submit-service.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  }
+
+  /**
+   * Used to register callbacks.
+   * @param {function(*)} cb
+   */
+  beforeSubmit(cb) {
```
Comment (lannka) [inline extensions/amp-form/0.1/form-submit-service.js]: this is good for now. we can separately introduce an async version later when needed.

File: extensions/amp-form/0.1/amp-form.js [hunk 3]
```diff
@@ -447,6 +451,7 @@ export class AmpForm {
    */
   submit_(trust) {
     const varSubsFields = this.getVarSubsFields_();
+    this.formSubmitService_.fire(this.element);
```
Comment (lannka) [inline extensions/amp-form/0.1/amp-form.js]: let's catch any exception and still let the form submit go through.

File: extensions/amp-form/0.1/form-submit-service.js (hunk 2 above)
Comment (calebcordry) [reply extensions/amp-form/0.1/form-submit-service.js]: SGTM

**Comment by calebcordry** [comment]
Changed to use `observerable.js` + other comments @cvializ @lannka PTAL.

**Pull request merged**